File: python/1b.py
# This is the bane of my existence btw 

pos = 50
count = 0

with open("C:/Users/saanv/Downloads/input.txt", "r") as lines:
  for line in lines:
    val = int(line[1:])
    if line[0] == "R":
      for i in range(val):
        pos += 1
        if pos == 100:
          pos = 0
          count += 1
    else:
      for i in range(val):
        pos -= 1
        if pos == -1:
          pos = 99
        if pos == 0:
          count += 1

print(count)

# attempt 1: 7004, it's not right, I think I was adding 1 to the range but that's actually not right
# attempt 2: 6932, I've never got that value before, but I still have to wait 3 mins 3 seconds, IT'S RIGHT LETS GOOOOOOOOO

File: java/1b.java
// this actually needs to be named b1.java, but I like having my files nice and neat

import java.io.*;
import java.util.*;

public class b1 {
  public static void main(String[] args) throws FileNotFoundException {
    int pos = 50;
    int count = 0;
    
    InputStream lines = new FileInputStream("C:/Users/saanv/Downloads/input.txt");
    Scanner input = new Scanner(lines);
    while (input.hasNextLine()) {
      String next = input.nextLine();
      int val = Integer.parseInt(next.substring(1));

      if (next.substring(0, 1).equals("R")) {
        for (int i = 0; i < val; i++) {
          pos++;
          if (pos == 100) {
            pos = 0;
            count++;
          }
        }
      } else {
        for (int i = 0; i < val; i++) {
          pos--;
          if (pos == -1) {
            pos = 99;
          }
          if (pos == 0) {
            count++;
          }
        }
      }
      
    }
    System.out.print(count);
  }
}
